mkosi/__init__.py

```diff
@@ -168,6 +170,36 @@ class MkosiException(Exception):
     """Leads to sys.exit"""
 
 
+def dictify(f: Callable[..., Generator[Tuple[str, str], None, None]]) -> Callable[..., Dict[str, str]]:
+    def wrapper(*args: Any, **kwargs: Any) -> Dict[str, str]:
+        return dict(f(*args, **kwargs))
+
+    return functools.update_wrapper(wrapper, f)
+
+
+@dictify
```

Comment:
Maybe we can just use a `dict` in `read_os_release` to store the results and return that? I think that ends up being a simpler solution overall.

Reply:
I prefer to keep the *interesting* part, i.e. `read_os_release()` as simple and clean as possible. `dictify` is something that you don't need to look at. As mentioned before, I hope to be able to reuse it in other places later.

Reply:
I think one can argue, whether moving the dictionary handling outside makes this simpler ;)
```python
def read_os_release() -> Dict[str, str]:
    osrelease = {}
    try:
        filename = "/etc/os-release"
        f = open(filename)
    except FileNotFoundError:
        filename = "/usr/lib/os-release"
        f = open(filename)

    for line_number, line in enumerate(f, start=1):
        line = line.rstrip()
        if not line or line.startswith("#"):
            continue
        m = re.match(r"([A-Z][A-Z_0-9]+)=(.*)", line)
        if m:
            name, val = m.groups()
            if val and val[0] in "\"'":
                val = ast.literal_eval(val)
            osrelease[name] = val
        else:
            print(f"{filename}:{line_number}: bad line {line!r}", file=sys.stderr)
    
    return osrelease
```
But I get you point and if it finds other users I don't see an issue with it.